pkg/binding/format/format.go

```diff
@@ -24,6 +24,14 @@ type Format interface {
 	Unmarshal([]byte, *ce.Event) error
 }
 
+// UnknownFormat allows an event with an unknown format string to be forwarded,
+// but Marshal() and Unmarshal will always fail.
+type UnknownFormat string
```

Comment:
Why did you added this new format? Any specific reason?

Reply:
Your changes pass a Format in places where we don't actually know if we can locally encode/decode it with a Format. Previously I had passed a string in such places, but I like the type hint of of passing a Format - so this is a special Format that allows you to pass a string through APIs that take a Format but fails if you actually try to encode or decode.